### PR TITLE
Refactor phase1

### PR DESCRIPTION
**Pull request by dieppa**

### CODE REVIEW

File: dimmer-local/src/main/java/com/github/cloudyrock/dimmer/FeatureConfigurationBuilder.java [hunk 1]
```diff
@@ -16,37 +14,63 @@
  * @see DimmerConfigException
  * @see DimmerFeature
  */
-final class FeatureConfigurationBuilder extends DimmerFeatureConfigurable<FeatureConfigurationBuilder>
+final class FeatureConfigurationBuilder extends DimmerFeatureConfigurable
         implements DimmerEnvironmentConfigurable<FeatureConfigurationBuilder> {
 
-    private static final DimmerLogger LOGGER = new DimmerLogger(FeatureConfigurationBuilder.class);
+    private static final DimmerLogger logger = new DimmerLogger(FeatureConfigurationBuilder.class);
```
Comment (dhivesh) [inline dimmer-local/src/main/java/com/github/cloudyrock/dimmer/FeatureConfigurationBuilder.java]: We should have static finals in upper case for consistency.

File: dimmer-local/src/main/java/com/github/cloudyrock/dimmer/FeatureConfigurationBuilder.java [hunk 2]
```diff
@@ -107,22 +131,257 @@ private FeatureExecutor getFeatureExecutor(EnvironmentConfig environmentConfig)
...
+                                   String operation,
+                                   Object valueToReturn) {
+        final FeatureMetadata metadata =
+                new ValueFeatureMetadata(feature, operation, valueToReturn);
+        addFeatureMetadata(metadata);
+        return newInstance(environments, configMetadata, defaultExceptionType, dimmerConfigReader);
+
+    }
+
+    private void addFeatureMetadata(FeatureMetadata metadata) {
```
Comment (dhivesh) [inline dimmer-local/src/main/java/com/github/cloudyrock/dimmer/FeatureConfigurationBuilder.java]: We could put this logic in a separate class in the future if it is growing to much

File: dimmer-local/src/main/java/com/github/cloudyrock/dimmer/FeatureConfigurationBuilder.java (hunk 1 above)
Comment (dieppa) [reply dimmer-local/src/main/java/com/github/cloudyrock/dimmer/FeatureConfigurationBuilder.java]: As discussed, we'll eventually move all the loggers to lower case, as we are following google standards.

File: dimmer-local/src/main/java/com/github/cloudyrock/dimmer/FeatureConfigurationBuilder.java (hunk 2 above)
Comment (dieppa) [reply dimmer-local/src/main/java/com/github/cloudyrock/dimmer/FeatureConfigurationBuilder.java]: we could ;)

**Pull request merged**